### PR TITLE
feat(macos): update spotify plugin to upstream

### PR DESCRIPTION
**Pull request by sobeckley**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.
- [x] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

This script is a copy/paste from https://github.com/hnarayanan/shpotify (the plugin's acknowledgements list it as such). The upstream script has been updated since this plugin was created, this PR pulls in those changes to this plugin.

## Other comments:

I noticed the version bundled in this plugin does not work with the latest version of Spotify, however the upstream does. Hopefully this will resolve that discrepancy.

I can't tell who (if anyone) is an active maintainer of this plugin so I'm not sure who to `@`, apologies for that.

### CODE REVIEW

File: plugins/macos/spotify [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env bash
 
-function spotify() {
```
Comment (carlosala) [inline plugins/macos/spotify:3]: This should not be removed, we are sourcing that on main plugin file.

File: plugins/macos/spotify [hunk 2]
```diff
@@ -148,11 +150,11 @@ if [ $# = 0 ]; then
 else
 	if [ ! -d /Applications/Spotify.app ] && [ ! -d $HOME/Applications/Spotify.app ]; then
 		echo "The Spotify application must be installed."
-		return 1
+		exit 1
```
Comment (carlosala) [inline plugins/macos/spotify:153]: We changed on purpose all exits to return, could you revert that?

File: plugins/macos/spotify [hunk 3]
```diff
@@ -307,7 +308,7 @@ while [ $# -gt 0 ]; do
 
         "quit"    ) cecho "Quitting Spotify.";
             osascript -e 'tell application "Spotify" to quit';
-            break ;;
```
Comment (carlosala) [inline plugins/macos/spotify:310]: Same here, it should be break.

File: plugins/macos/spotify [hunk 4]
```diff
@@ -468,11 +469,9 @@ while [ $# -gt 0 ]; do
...
             break ;;
-
         * )
             showHelp;
-            return 1 ;;
+            exit 1;
 
     esac
 done
-}
```
Comment (carlosala) [inline plugins/macos/spotify:478]: This should not be removed.

**Review by carlosala — CHANGES_REQUESTED** [review]

**Comment by sobeckley** [comment]
@carlosala Apologies for the extended delay. This should be ready for another look whenever you have a moment.

**Comment by carlosala** [comment]
Thanks!

**Pull request merged**